go-eicio/reader.go

```diff
@@ -3,46 +3,59 @@ package eicio
 import (
 	"encoding/binary"
 	"io"
+	"os"
 )
 
 type Reader struct {
 	byteReader io.Reader
 }
 
+func Open(filename string) (*Reader, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	return NewReader(file), nil
+}
+
+func (rdr *Reader) Close() {
+	rdr.byteReader.(*os.File).Close()
+}
+
 func NewReader(byteReader io.Reader) *Reader {
 	return &Reader{
 		byteReader: byteReader,
 	}
 }
 
-func (rdr *Reader) GetEvent() (event *Event, err error) {
+func (rdr *Reader) GetEvent() *Event {
```

Comment:
I would just keep the `(*Event, error)` output values.
and let users handle the error as they see fit.

possibly, I would "decorate" the error using [github.com/pkg/errors](https://godoc.org/github.com/pkg/errors).

another (orthogonal) idea would be to add to `Reader` itself a slice of "recovery functions" that could kick off when a special magic number is encountered and (try to) rewind the stream/resynchronize.

but I think I would still return a `(*Event, error)`.